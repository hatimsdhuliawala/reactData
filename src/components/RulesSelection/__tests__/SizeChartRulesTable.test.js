import ShallowRenderer from 'react-test-renderer/shallow'
import RulesTable from '../Components/SizeChartRulesTable'
import React from 'react'

const topLevelRules = [
  {
    category: 'baby-bedding',
    brand: '',
    size: '',
    items: 2395,
    rules: 9,
  },
  {
    category: 'baby-travel',
    brand: '',
    size: '',
    priority: undefined,
    items: 1198,
    rules: 11,
  },
  {
    category: 'boys-toddler-sleepwear',
    brand: '',
    size: '',
    priority: undefined,
    items: 1984,
    rules: 27,
  },
  {
    category: 'electronics wearable tech',
    brand: '',
    size: '',
    priority: undefined,
    items: 98,
    rules: 12,
  },
  {
    category: 'girls-toddler-outerwear',
    brand: '',
    size: '',
    priority: undefined,
    items: 290,
    rules: 5,
  },
  {
    category: 'girls-toddler-uniforms',
    brand: '',
    size: '',
    priority: undefined,
    items: 154,
    rules: 2,
  },
  {
    category: 'home-bedding',
    brand: '',
    size: '',
    priority: undefined,
    items: 456,
    rules: 15,
  },
  {
    category: 'pet-apparel-costumes',
    brand: '',
    size: '',
    priority: undefined,
    items: 102,
    rules: 4,
  },
  {
    category: 'men-blazers-jackets',
    brand: '',
    size: '',
    priority: undefined,
    items: 88,
    rules: 11,
  },
  {
    category: 'men-shirts',
    brand: '',
    size: '',
    priority: undefined,
    items: 158,
    rules: 8,
  },
  {
    category: 'women-activewear',
    brand: '',
    size: '',
    priority: undefined,
    items: 1005,
    rules: 36,
  },
  {
    category: 'women-dresses',
    brand: '',
    size: '',
    priority: undefined,
    items: 3145,
    rules: 95,
  },
  {
    category: 'women-jumpsuits-rompers',
    brand: '',
    size: '',
    priority: undefined,
    items: 302,
    rules: 4,
  },
]

describe('Rules Table', () => {
  let renderer

  beforeEach(() => {
    renderer = new ShallowRenderer()
  })

  it('Should Render the top level table', () => {
    const tree = renderer.render(
      <RulesTable
        rules={topLevelRules}
      />
    )
    expect(tree).toMatchSnapshot('Rules Table Top Level')
  })
})
