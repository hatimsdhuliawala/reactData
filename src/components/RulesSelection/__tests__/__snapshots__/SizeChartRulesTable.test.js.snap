// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules Table Should Render the top level table: Rules Table Top Level 1`] = `
<RulesTable
  classes={
    Object {
      "NewSizeChartButton": "RulesTable-NewSizeChartButton-5",
      "RulesBuilderControl": "RulesTable-RulesBuilderControl-17",
      "RulesBuilderControlButton": "RulesTable-RulesBuilderControlButton-18",
      "RulesBuilderHeader": "RulesTable-RulesBuilderHeader-16",
      "RulesBuilderMain": "RulesTable-RulesBuilderMain-15",
      "RulesBuilderStepButton": "RulesTable-RulesBuilderStepButton-19",
      "RulesBuilderStepper": "RulesTable-RulesBuilderStepper-14",
      "RulesMain": "RulesTable-RulesMain-4",
      "RulesSelection": "RulesTable-RulesSelection-1",
      "RulesSelectionFilter": "RulesTable-RulesSelectionFilter-3",
      "RulesSelectionHeader": "RulesTable-RulesSelectionHeader-2",
      "expandedRulesTableCard": "RulesTable-expandedRulesTableCard-12",
      "hidden": "RulesTable-hidden-21",
      "rulesBuilderContent": "RulesTable-rulesBuilderContent-20",
      "rulesExpansionPanelChild": "RulesTable-rulesExpansionPanelChild-11",
      "rulesListTableListWrapper": "RulesTable-rulesListTableListWrapper-41",
      "rulesTCINTable": "RulesTable-rulesTCINTable-35",
      "rulesTCINTableBody": "RulesTable-rulesTCINTableBody-29",
      "rulesTCINTableCCCell": "RulesTable-rulesTCINTableCCCell-34",
      "rulesTCINTableChart": "RulesTable-rulesTCINTableChart-24",
      "rulesTCINTableFilter": "RulesTable-rulesTCINTableFilter-28",
      "rulesTCINTableImage": "RulesTable-rulesTCINTableImage-22",
      "rulesTCINTableImageCell": "RulesTable-rulesTCINTableImageCell-30",
      "rulesTCINTableLDCell": "RulesTable-rulesTCINTableLDCell-33",
      "rulesTCINTableSummary": "RulesTable-rulesTCINTableSummary-27",
      "rulesTCINTableTCIN": "RulesTable-rulesTCINTableTCIN-25",
      "rulesTCINTableTCINCell": "RulesTable-rulesTCINTableTCINCell-32",
      "rulesTCINTableTitle": "RulesTable-rulesTCINTableTitle-23",
      "rulesTCINTableTitleCell": "RulesTable-rulesTCINTableTitleCell-31",
      "rulesTCINTableTop": "RulesTable-rulesTCINTableTop-26",
      "rulesTableDataText": "RulesTable-rulesTableDataText-9",
      "rulesTableDataTitleText": "RulesTable-rulesTableDataTitleText-10",
      "rulesTableDeleteDiv": "RulesTable-rulesTableDeleteDiv-13",
      "rulesTableHeader": "RulesTable-rulesTableHeader-40",
      "rulesTableHeaderText": "RulesTable-rulesTableHeaderText-8",
      "rulesTableIdentifiers": "RulesTable-rulesTableIdentifiers-6",
      "rulesTableList": "RulesTable-rulesTableList-39",
      "rulesTableListIcon": "RulesTable-rulesTableListIcon-38",
      "rulesTableListMain": "RulesTable-rulesTableListMain-36",
      "rulesTableListSecondaryButton": "RulesTable-rulesTableListSecondaryButton-37",
      "rulesTableStats": "RulesTable-rulesTableStats-7",
    }
  }
  deletePressed={[Function]}
  editRule={[Function]}
  expendRule={[Function]}
  rules={
    Array [
      Object {
        "brand": "",
        "category": "baby-bedding",
        "items": 2395,
        "rules": 9,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "baby-travel",
        "items": 1198,
        "priority": undefined,
        "rules": 11,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "boys-toddler-sleepwear",
        "items": 1984,
        "priority": undefined,
        "rules": 27,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "electronics wearable tech",
        "items": 98,
        "priority": undefined,
        "rules": 12,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "girls-toddler-outerwear",
        "items": 290,
        "priority": undefined,
        "rules": 5,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "girls-toddler-uniforms",
        "items": 154,
        "priority": undefined,
        "rules": 2,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "home-bedding",
        "items": 456,
        "priority": undefined,
        "rules": 15,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "pet-apparel-costumes",
        "items": 102,
        "priority": undefined,
        "rules": 4,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "men-blazers-jackets",
        "items": 88,
        "priority": undefined,
        "rules": 11,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "men-shirts",
        "items": 158,
        "priority": undefined,
        "rules": 8,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "women-activewear",
        "items": 1005,
        "priority": undefined,
        "rules": 36,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "women-dresses",
        "items": 3145,
        "priority": undefined,
        "rules": 95,
        "size": "",
      },
      Object {
        "brand": "",
        "category": "women-jumpsuits-rompers",
        "items": 302,
        "priority": undefined,
        "rules": 4,
        "size": "",
      },
    ]
  }
/>
`;
